pkg/adapter/adapter.go

```diff
@@ -452,15 +476,20 @@ func (a *Adapter) ExecuteTxs(
 		PartSetHeader: cmttypes.PartSetHeader{Total: 1, Hash: block.DataHash},
 	}
 
-	if err := fireEvents(a.EventBus, block, currentBlockID, fbResp, validatorUpdates); err != nil {
-		a.Logger.Error("failed to fire events", "err", err)
+	// save the finalized block response
+	if err := a.Store.SaveBlockResponse(ctx, blockHeight, fbResp); err != nil {
+		return nil, 0, fmt.Errorf("failed to save block response: %w", err)
+	}
+	a.stackBlockCommitEvents(currentBlockID, block, fbResp, validatorUpdates)
```

Comment:
first stack the events so that they can be released when soft consensus is reached